### PR TITLE
Change and fix serialization of OLF parameters to map

### PR DESCRIPTION
**Pull request by obrix**
**Please check if the PR fulfills these requirements**

- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] A PR or issue has been opened in all impacted repositories (if any)

**Does this PR already have an issue describing the problem?**

No

**What kind of change does this PR introduce?**

Change and fix serialization of OpenLoadFlowParameters to map so the retrieved map can be reused to initialize another OpenLoadFlowParameters.

**What is the current behavior?**

Retrieved map representation when used with  OpenLoadFlowProvider.updateSpecificParameters lead to a crash because  serialized empty list and set properties cannot be deserialized.
Also null string parameters are serialized as empty string 

**What is the new behavior (if this is a feature change)?**

Output map can now be reused as an input of updateSpecificParameters function. List and Set parameters are serialized to String properties without "[" and "]". 
Empty list or set are serialized as empty string which is correctly handled by deserialization process.
There was two parameters with null default value : debugDir and OuterLoopNames. Default value was changed to empty string and empty list and associated semantic has been updated.

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [ ] No

**If yes, please check if the following requirements are fulfilled**

- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration steps are described in the following section

**What changes might users need to make in their application due to this PR? (migration steps)**

**Other information**:

### CODE REVIEW

File: src/main/java/com/powsybl/openloadflow/OpenLoadFlowParameters.java [hunk 1]
```diff
@@ -1995,15 +1995,15 @@ public static AcLoadFlowParameters createAcParameters(Network network, LoadFlowP
 
     public static List<AcOuterLoop> createAcOuterLoops(LoadFlowParameters parameters, OpenLoadFlowParameters parametersExt) {
         AcOuterLoopConfig outerLoopConfig = AbstractAcOuterLoopConfig.getOuterLoopConfig()
-                .orElseGet(() -> parametersExt.getOuterLoopNames() != null ? new ExplicitAcOuterLoopConfig()
-                                                                           : new DefaultAcOuterLoopConfig());
+                .orElseGet(() -> parametersExt.getOuterLoopNames() == null || parametersExt.getOuterLoopNames().isEmpty()
+                    ? new DefaultAcOuterLoopConfig() : new ExplicitAcOuterLoopConfig());
```
[the remark this answers is not in the document]
Comment (jeandemanged) [reply src/main/java/com/powsybl/openloadflow/OpenLoadFlowParameters.java:1999]: this is indeed an edge case of `ExplicitAcOuterLoopConfig` for which no unit test is today present.

File: src/test/java/com/powsybl/openloadflow/OpenLoadFlowParametersTest.java [hunk 2]
```diff
@@ -353,6 +353,24 @@ void updateEmptyStringListParametersIssue() {
...
+        var map = provider.createMapFromSpecificParameters(parameters.getExtension(OpenLoadFlowParameters.class));
+
+        LoadFlowParameters parameters2 = new LoadFlowParameters();
+        parameters2.addExtension(OpenLoadFlowParameters.class, new OpenLoadFlowParameters());
+        //try reupdating parameters with map
+        provider.updateSpecificParameters(parameters2.getExtension(OpenLoadFlowParameters.class), map);
+
+        assertTrue(OpenLoadFlowParameters.equals(parameters, parameters2));
+    }
+
```
Comment (vidaldid-rte) [inline src/test/java/com/powsybl/openloadflow/OpenLoadFlowParametersTest.java:373]: What about adding a map serializatoin/deserialization test in the testEqualsCloneAndUpdate test to cover all parameters ?
Note: for lists, this test (testEqualsCloneAndUpdate) does not cover empty lists. It could be extended.

File: src/main/java/com/powsybl/openloadflow/OpenLoadFlowProvider.java [hunk 3]
```diff
@@ -333,9 +332,20 @@ public Optional<Class<? extends Extension<LoadFlowParameters>>> getSpecificParam
...
+        Map<String, String> resMap = new HashMap<>();
+        parametersMap.forEach((key, value) -> {
+            if (value != null) {
+                if (value instanceof List || value instanceof Set) {
+                    resMap.put(key, String.join(",", ((Collection<?>) value).stream().map(Object::toString).toList()));
+                } else {
+                    resMap.put(key, Objects.toString(value));
+                }
+            } else {
+                resMap.put(key, null);
```
[the remark this answers is not in the document]
Comment (vidaldid-rte) [reply src/main/java/com/powsybl/openloadflow/OpenLoadFlowProvider.java:345]: I think it's OK. 
What matters for me is that serialization + reading is idempotent (ie gives same as original, including when original had a null param value, for example for the explicit outerloops). That's why I think it would be safer if we had a test for all parameters and relevant type of values. Safer for now, and when new parameters are introduced.

File: src/test/java/com/powsybl/openloadflow/OpenLoadFlowParametersTest.java (hunk 2 above)
Comment (obrix) [reply src/test/java/com/powsybl/openloadflow/OpenLoadFlowParametersTest.java:375]: Looks good to me thanks.

**Review by vidaldid-rte — APPROVED** [review]
